drivers/power_gating/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2022 Intel Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_POWER_GATING_INTEL power_gating_intel.c)
```

Comment:
Is there any reason this is not implemented in `drivers/power_domain/` ? Also, it should follow the power_domain nomenclature.

Reply:
Moved and renamed to power_domain_intel

Reply:
Should this be more specific to `_intel_cavs` or `_intel_audio`? These are not universal to all IP blocks on Intel hardware.

Reply:
Sorry @dcpleung, missed Your comment here. Changed to `_intel_adsp`. I'd like to avoid `_cavs` naming scheme due to upcoming ace platform.